uint256.go

```diff
@@ -1209,3 +1209,38 @@ func (z *Int) ExtendSign(x, byteNum *Int) *Int {
 	}
 	return z
 }
+
+// Sqrt sets z to ⌊√x⌋, the largest integer such that z² ≤ x, and returns z.
+func (z *Int) Sqrt(x *Int) *Int {
+	// This implementation of Sqrt is based on big.Int (see math/big/nat.go).
+	if x.LtUint64(2) {
+		return z.Set(x)
+	}
+	var (
+		z1   = &Int{1, 0, 0, 0}
+		z2   = &Int{}
+		a, b uint64
+	)
+	// Start with value known to be too large and repeat "z = ⌊(z + ⌊x/z⌋)/2⌋" until it stops getting smaller.
+	z1 = z1.Lsh(z1, uint(x.BitLen()+1)/2) // must be ≥ √x
+	for {
+		z2 = z2.Div(x, z1)
+		z2 = z2.Add(z2, z1)
+		{ //z2 = z2.Rsh(z2, 1) -- the code below does a 1-bit rsh faster
```

Comment:
There is DeepSource complain about this additional block.

Reply:
Yeah deepsource is nitpicky. "Unnecessary block detected" - hmpf. 
Do you also think it should be removed? I personally like partitioning stuff like that sometimes, for clarity.

Reply:
I could move the `a,b` definition inside this clause to make deepsource happy

Reply:
> I could move the a,b definition inside this clause to make deepsource happy

This is good idea.